src/nvim/screen.c

```diff
@@ -504,7 +512,7 @@ void update_single_line(win_T *wp, linenr_T lnum)
   int j;
 
   // Don't do anything if the screen structures are (not yet) valid.
-  if (!screen_valid(true) || updating_screen) {
+  if (linebuf_char == NULL || updating_screen) {
```

Comment:
What is the reason for this change? Note `screen_valid` can also allocate the global grid in some situations if it wasn't already (though I don't know if it is still used).